giskard/llm/talk/utils/shap.py

```diff
@@ -0,0 +1,91 @@
+from typing import TYPE_CHECKING
+
+import numpy as np
+
+if TYPE_CHECKING:
+    from giskard import Dataset
+    from giskard.models.base import BaseModel
+
+from giskard.core.errors import GiskardImportError
+
+
+def _calculate_shap_values(data_to_explain: "Dataset", model: "BaseModel", dataset: "Dataset") -> np.ndarray:
+    """Perform Kernel SHAP explanation.
+
+    Returns the feature importance in terms of SHAP values. Kernel SHAP is used
+    to calculate feature contributions.
+
+    Parameters
+    ----------
+    data_to_explain : Dataset
+        The data for the prediction to explain.
+    model : giskard.Model
+        The model used for the SHAP explanation.
+    dataset : giskard.Dataset
+        The dataset used for the SHAP explanation.
+
+    Returns
+    -------
+    shap_values : np.ndarray
+        The model's SHAP values as a numpy array
+    """
+    from giskard.models.model_explanation import _get_background_example, _prepare_for_explanation
+
+    try:
+        from shap import KernelExplainer
+    except ImportError as e:
+        raise GiskardImportError("shap") from e
```

Comment:
Isn't this a basic deps from our lib ? is this case possible ?

Reply:
@rabah-khalek What do you think?

Reply:
@Hartorn the SHAP is not the part of basic deps, but of the [hub] flavour. Also, I will create the new group [talk], where I will put necessary packages, including 'ship'

Reply:
Leave it, cause the default deps do not contain SHAP package.